src/savedata.c

```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
+    sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, !saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
+    sub_0202516C(saveData, SAVE_BLOCK_ID_NORMAL, saveData->unk_20020[SAVE_BLOCK_ID_NORMAL]);
+    sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
 
-    MI_CpuFillFast(v2, 0xffffffff, 0x1000);
+    MI_CpuFillFast(saveBuffer, 0xffffffff, SAVE_SECTOR_SIZE);
 
-    for (v0 = 0; v0 < 32 * 2; v0++) {
-        sub_02025A9C(0x1000 * (v0 + 0), v2, 0x1000);
-        sub_02025A9C(0x1000 * (v0 + 64), v2, 0x1000);
+    for (sectorIndex = 0; sectorIndex < SAVE_PAGE_MAX * 2; sectorIndex++) {
+        sub_02025A9C(SAVE_SECTOR_SIZE * (sectorIndex + 0), saveBuffer, SAVE_SECTOR_SIZE);
+        sub_02025A9C(SAVE_SECTOR_SIZE * (sectorIndex + 64), saveBuffer, SAVE_SECTOR_SIZE);
     }
 
-    Heap_FreeToHeap(v2);
-    sub_0202479C(param0);
+    Heap_FreeToHeap(saveBuffer);
+    SaveData_Clear(saveData);
 
-    param0->unk_04 = 0;
+    saveData->dataExists = FALSE;
     SleepUnlock(1);
 
-    return 1;
+    return TRUE;
 }
 
-BOOL sub_020246A8 (SaveData * param0)
+BOOL SaveData_Load (SaveData *saveData)
 {
-    BOOL v0;
+    BOOL loadSucceeded;
 
-    if (!param0->unk_00) {
-        return 0;
+    if (!saveData->backupExists) {
+        return FALSE;
     }
 
-    v0 = sub_02024DBC(param0);
+    loadSucceeded = sub_02024DBC(saveData);
 
-    if (v0) {
-        param0->unk_04 = 1;
-        param0->unk_08 = 0;
+    if (loadSucceeded) {
+        saveData->dataExists = TRUE;
+        saveData->isNewGameData = FALSE;
 
         {
-            int v1, v2;
+            int frontierResult, videoResult;
 
-            sub_02024CD4(param0, &v1, &v2);
+            SaveDataExtra_LoadCheck(saveData, &frontierResult, &videoResult);
         }
 
-        return 1;
+        return TRUE;
     } else {
-        return 0;
+        return FALSE;
     }
 }
 
-int sub_020246E0 (SaveData * param0)
+int SaveData_Save (SaveData *saveData)
 {
-    int v0;
+    int saveResult;
 
-    if (!param0->unk_00) {
-        return 3;
+    if (!saveData->backupExists) {
+        return SAVE_RESULT_CORRUPT;
     }
 
-    if (param0->unk_08) {
+    if (saveData->isNewGameData) {
         SleepLock(1);
 
-        sub_0202516C(param0, 0, !param0->unk_20020[0]);
-        sub_0202516C(param0, 1, !param0->unk_20020[1]);
-        sub_0202516C(param0, 0, param0->unk_20020[0]);
-        sub_0202516C(param0, 1, param0->unk_20020[1]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_NORMAL, !saveData->unk_20020[SAVE_BLOCK_ID_NORMAL]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, !saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_NORMAL, saveData->unk_20020[SAVE_BLOCK_ID_NORMAL]);
+        sub_0202516C(saveData, SAVE_BLOCK_ID_BOXES, saveData->unk_20020[SAVE_BLOCK_ID_BOXES]);
 
         SleepUnlock(1);
     }
 
-    v0 = sub_0202513C(param0);
+    saveResult = sub_0202513C(saveData);
```

Comment:
Continuation from the above:

```suggestion
    int saveResult = sub_0202513C(saveData);
```